plugins/inputs/systemd_units/systemd_units_linux.go

```diff
@@ -150,6 +151,10 @@ func (s *SystemdUnits) Gather(acc telegraf.Accumulator) error {
 			acc.AddError(fmt.Errorf("Error parsing line (expected at least 4 fields): %s", line))
 			continue
 		}
+		// not found or failed units might have a leading asterisk
+		if data[0] == "*" {
+			data = data[1:]
+		}
```

Comment:
The problem is that we remove things _after_ the length check. There might be cases where you get four fields, but an asterix in the beginning and then end up with only three fields left in _data_.

How about adding
```suggestion
if strings.HasPrefix(line, "*") {
     line = strings.replace(line, "*", "", 1)
}
```
above, directly after the assignment to line (line 148 I think) and **before** creating _data_?